superset-frontend/packages/superset-ui-chart-controls/src/sections/timeComparison.tsx

```diff
@@ -94,9 +98,29 @@ export const timeComparisonControls: ({
         name: 'start_date_offset',
         config: {
           type: 'TimeOffsetControl',
-          label: t('shift start date'),
+          label: t('Shift start date'),
           visibility: ({ controls }) =>
             controls?.time_compare.value === 'custom',
+          mapStateToProps: (
+            state: ControlPanelState,
+            controlState: ControlState,
+          ) => {
+            const { form_data } = state;
+            const { time_compare } = form_data;
+            const newState = { ...controlState };
+            if (
+              time_compare === 'custom' &&
+              (controlState.value === '' ||
+                controlState.value === 'Invalid date')
```

Comment:
We use „Invalid date” here and in explore. Should we move it to a const in chart controls package? Also, should it be translated or does that string come from some external library?

Reply:
Let me move that to a const, about translating, that's coming form moment indeed.

Reply:
Can we write a small unit test for that „Invalid date” string to future proof it in case it ever changes in momentjs?

Reply:
Sure thing